cmd/cluster-agent-cloudfoundry/subcommands/run/command.go

```diff
@@ -139,12 +143,11 @@ func run(log log.Component, taggerComp tagger.Component, demultiplexer demultipl
 
 	// create and setup the Autoconfig instance
 	// The Autoconfig instance setup happens in the workloadmeta start hook
-	// create and setup the Collector and others.
-	common.LoadComponents(demultiplexer, secretResolver, wmeta, pkgconfig.Datadog.GetString("confd_path"))
+	common.LoadComponents(secretResolver, wmeta, pkgconfig.Datadog.GetString("confd_path"))
 
 	// Set up check collector
-	common.AC.AddScheduler("check", collector.InitCheckScheduler(common.Coll, demultiplexer), true)
-	common.Coll.Start()
+	common.AC.AddScheduler("check", pkgcollector.InitCheckScheduler(optional.NewOption[pkgcollector.Collector](collector), demultiplexer), true)
+	collector.Start()
 
```

Comment:
You may miss to initialize diagnose package.

Reply:
I'll double check this one thanks for the heads up! It was difficult to determine all the places the diagnose package needed to be initialized. Glad to hear you are already working on this though!